cpp/src/parquet/column_writer.cc

```diff
@@ -1741,15 +1752,22 @@ class TypedColumnWriterImpl : public ColumnWriterImpl,
     if (num_values != num_spaced_values) {
       current_value_encoder_->PutSpaced(values, static_cast<int>(num_spaced_values),
                                         valid_bits, valid_bits_offset);
+      UpdateBloomFilterSpaced(values, num_spaced_values, valid_bits, valid_bits_offset);
     } else {
       current_value_encoder_->Put(values, static_cast<int>(num_values));
+      UpdateBloomFilter(values, num_values);
     }
     if (page_statistics_ != nullptr) {
       page_statistics_->UpdateSpaced(values, valid_bits, valid_bits_offset,
                                      num_spaced_values, num_values, num_nulls);
     }
     UpdateUnencodedDataBytes();
   }
+
+  void UpdateBloomFilter(const T* values, int64_t num_values);
```

Comment:
Does it make sense to move all these `UpdateXXX` to the `BloomFilter` class? These functions look more like the detail of bloom filter instead of a column writer.